pkg/util/issutil/leaderselectionpolicy.go

```diff
@@ -54,3 +85,184 @@ func (simple *SimpleLeaderPolicy) Reconfigure(nodeIDs []t.NodeID) LeaderSelectio
+// if this one is more recent than the one it already has
+func (l *BlacklistLeaderPolicy) Suspect(e t.EpochNr, node t.NodeID) {
+	if _, ok := l.Membership[node]; !ok { //node is a not a member
+		//TODO error but cannot be passed through
+		return
+	}
+
+	if epochNr, ok := l.Suspected[node]; !ok || epochNr < e {
+		l.Suspected[node] = e // update with latest epoch that node was suspected
+	}
+}
+
+// Reconfigure informs the leader selection policy about a change in the membership.
+func (l *BlacklistLeaderPolicy) Reconfigure(nodeIDs []t.NodeID) LeaderSelectionPolicy {
+	membership := make(map[t.NodeID]struct{}, len(nodeIDs))
+	suspected := make(map[t.NodeID]t.EpochNr, len(nodeIDs))
+	for _, nodeID := range nodeIDs {
+		if sus, ok := l.Membership[nodeID]; ok {
+			membership[nodeID] = sus // keep former suspect as suspected
+			if epoch, ok := l.Suspected[nodeID]; ok {
+				suspected[nodeID] = epoch
+			}
+		} else {
+			membership[nodeID] = struct{}{}
+		}
+	}
+
+	newPolicy := BlacklistLeaderPolicy{
+		membership,
+		suspected,
+		l.minLeaders,
+	}
+
+	return &newPolicy
+}
+
+type BlacklistLeaderPolicySerialization struct {
+	Membership []keyValue[t.NodeID, struct{}]
+	Suspected  []keyValue[t.NodeID, t.EpochNr]
+	MinLeaders int
+}
+
+func (l *BlacklistLeaderPolicy) Bytes() ([]byte, error) {
+	membership := mapToSortedSlice[t.NodeID, struct{}](l.Membership)
+	suspected := mapToSortedSlice[t.NodeID, t.EpochNr](l.Suspected)
+	ser, err := cbor.Marshal(BlacklistLeaderPolicySerialization{
+		membership,
+		suspected,
+		l.minLeaders,
+	})
+	if err != nil {
+		return nil, err
+	}
+	out := t.Uint64ToBytes(uint64(Blacklist))
+	out = append(out, ser...)
+	return out, nil
+}
+
+func BlacklistLeaderPolicyFromBytes(data []byte) (*BlacklistLeaderPolicy, error) {
+	bser := &BlacklistLeaderPolicySerialization{}
+	if err := cbor.Unmarshal(data, bser); err != nil {
+		return nil, err
+	}
+
+	// Convert slices of key-value structs back to maps
+	b := &BlacklistLeaderPolicy{}
+	b.Membership = sliceToMap[t.NodeID, struct{}](bser.Membership)
+	b.Suspected = sliceToMap[t.NodeID, t.EpochNr](bser.Suspected)
+	b.minLeaders = bser.MinLeaders
+	return b, nil
+}
+
+// keyValue is a struct that holds a key-value pair of a map
+type keyValue[K constraints.Ordered, V any] struct {
+	Key   K
+	Value V
+}
+
+func mapToSortedSlice[K constraints.Ordered, V any](m map[K]V) []keyValue[K, V] {
+	keys := make([]K, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+
+	sort.Slice(keys, func(i, j int) bool {
+		// Sort keys to ensure deterministic order
+		return keys[i] < keys[j]
+	})
+
+	slice := make([]keyValue[K, V], len(keys))
+	for i, key := range keys {
+		slice[i] = keyValue[K, V]{
+			Key:   key,
+			Value: m[key],
+		}
+	}
+	return slice
+}
+
+func sliceToMap[K constraints.Ordered, V any](slice []keyValue[K, V]) map[K]V {
+	m := make(map[K]V, len(slice))
+	for _, kv := range slice {
+		m[kv.Key] = kv.Value
+	}
+	return m
+}
```

Comment:
If you choose to adopt the above suggestion, this becomes obsolete too.

Reply:
Gone with the new CBOR encmode (see other comment)